### ISSUE TITLE
Fix Checkbox & Radio empty `label` elements

### ISSUE THREAD

**Issue by mimarz** [digdir/designsystemet#1735]
Our current implementation has an empty `label` element if no label text is provided, causing errors in `a11y` checkers. 

This is due to styling of `Checkbox` done with `label::before` selector and hiding `input` element.

Explore if we can solve this better by styling `input` instead or provide a `hideLabel` prop. 

Notes: 
- Discussion from Slack: https://designsystemet.slack.com/archives/C05RF86A3K7/p1711034279790019
```[tasklist]
### Tasks
- [ ] https://github.com/digdir/designsystemet/issues/1740
- [ ] https://github.com/digdir/designsystemet/issues/1741
```

**Comment by mimarz** [comment]
Solved!
